### ISSUE TITLE
update sizePerPage through componentWillReceiveProps does not update select list

### ISSUE THREAD

**Issue by Raffiki** [AllenFang/react-bootstrap-table#659]
I have the following table. 

`SizePerPage` defaults to `10` but should be updated through `componentWillReceiveProps`.

If this happens and `options.sizePerTable` becomes say `30`, 30 rows are indeed rendered, but the `sizePerPage` widget still shows `10`.
[of interest: on the next `render`, by opening another component and coming back to this one, the widget updates to the desired `30`]

Is there anything obvious I am overlooking here? Do I pass the correct parameters?

```
export default class PaginationTable extends React.Component {
  constructor (props) {
    super(props);
    this.state = {
      options: {
        page: this.props.data.options._page || 1,  // which page you want to show as default
        onPageChange: this.props.onPageChange,
        onFilterChange: this.props.onFilterChange,
        onRowClick: this.props.onRowClick,
        sizePerPage: this.props.data.options._pageSize || 10,  // which size per page you want to locate as default
        pageStartIndex: 1, // where to start counting the pages
        paginationSize: 9,  // the pagination bar size.
        prePage: '<', // Previous page button text
        nextPage: '>', // Next page button text
        firstPage: '<<', // First page button text
        lastPage: '>>' // Last page button text
      }
    };
  }

  componentWillReceiveProps (newProps) {
    const sizePerPage = _.get(newProps, 'data.options._pageSize');
    if (sizePerPage !== this.props.data.options._pageSize) {
      this.setState({
        options: _.assign({}, this.state.options, {sizePerPage})
      });
    }
  }

  render () {
    return (
      <BootstrapTable
        data={this.props.data.data}
        fetchInfo={{ dataTotalSize: this.props.data.numRows }}
        remote={true}
        pagination={true}
        options={this.state.options}
        striped={true}
        hover={true}
        selectRow={this.props.selectRow}>
        {this.props.children}
      </BootstrapTable>
    );
  }
```

**Comment by Raffiki** [comment]
It is on the next `mount` (vs. render) of the component, that the correct `sizePerPage` is shown. So my 2cnts is `sizePerPage` is not picked up at `componentWillReceiveProps` on the react-bootstrap-table side. Need to doublecheck this though

**Comment by Raffiki** [comment]
adding `sizePerPage` to state when `remote` is true seems to fix this.
https://github.com/AllenFang/react-bootstrap-table/blob/master/src/BootstrapTable.js#L173
Any reason why this is explicitly left out?

**Comment by AllenFang** [comment]
@Raffiki, sorry for lately reply, I'll check this out

**Comment by AllenFang** [comment]
@Raffiki, if you want to handle any change by yourself, like page change, insert row, sort change, filter change, you should give ```remote``` true.

Actually, ```remote``` means you have the data which controlled by you store(flux) or something like that. ```remote``` default is false, ```react-bootstrap-table``` will handle all the state change inside.

You can check [this](https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/remote/remote-store-paging.js) example to see how to handle pagination on ```remote``` mode.

Feel free to discuss with me, I'll open this issue if this problem is still remain

**Comment by Raffiki** [comment]
@AllenFang I already do the things you say (paging, filtering etc) with `remote` set to `true`. This works fine. So, unless I'm missing something, that's not really my issue.

What I'm presumably missing, is a way of updating `sizePerPage` through `componentWillReceiveProps` in this setup. In fact, it seems to be omitted deliberatly: https://github.com/AllenFang/react-bootstrap-table/blob/master/src/BootstrapTable.js#L171-L174

I'm wondering why, as adding `sizePerPage` to the mentioned code block solves my issue.

**Comment by AllenFang** [comment]
ok, I'll firgure it

**Comment by doshisunny** [comment]
@AllenFang
I am having also same issue and I tested with Raffiki solution. 
And adding **sizePerPage:nextProps.options.sizePerPage** resolves the issue also for me.
this seems like missing from set state.
Let me know if you want me to make pull request.

**Comment by AllenFang** [comment]
@doshisunny, thanks, PR is welcome, because I'm busy on my work ;(  But I think I can fix it and release in this weekend

**Comment by doshisunny** [comment]
@AllenFang 
I understand you are busy. 
I created PR for this. Please merge it with your next release.

**Comment by AllenFang** [comment]
@doshisunny, thank 👍   I'll check this out

**Comment by AllenFang** [comment]
Merged #672. Released on ```v2.5.1```
